### DISCUSSION TITLE
Copy-DbaAgentJob -DisableOnDestination can fail to disable

### DISCUSSION THREAD

**Discussion by Hopeandfear** [dataplat/dbatools#8461] [Q&A] [answered]
Hi, 

Copy-DbaAgentJob -DisableOnDestination can fail, creating the job but not disabling the job, anyway to create the job already disabled rather than copying over enabled first?

I have tried working around this issue with getting jobs exporting to a SQL file and then running Invoke-SqlCmd or Invoke-DbaQuery to run the file but this cause a lot of failures handling big jobs with various USE / GO and quotation errors. 
This is much messier as I have to script out the job, use an replace command to change any @enabled=1 to @enabled=0 (which then also disables the schedule, as regex replacement cause further corruption)

For safety I need to ensure I do not have any jobs copied to the destination (secondary) server in an enabled state

I can't seem to see a way to copy the job in a disabled state first (I have been instructed that I can't disable on source and copy then re-enable on source afterwards, although not even sure if that would have worked but haven't tested this given my instructions to not amend the jobs on source.

Any ideas?

**Comment by potatoqualitee** [comment, marked as answer]
Hmmm perhaps the originating command can be modified to modify the script with a replace prior to migration. But using Export-DbaScript worked for me, you can try it in the interim.

![image](https://user-images.githubusercontent.com/8278033/180825752-dc4733ae-5b88-459e-a8b8-39b8d3cf1bb5.png)

**Comment by Hopeandfear** [comment]
Hi, thanks for such a quick reply, and my apologies for not replying sooner.

I did proceed down the path of scripting out the jobs individually to run these as script files to create the jobs on the secondary (destination) server, which to be honest had less errors with the job step codes using Invoke-Sqlcmd versus Invoke-DbaQuery, which threw up way more errors with syntax issues that when parsed in SQL where fine, thing GO and other commands threw it out... (but nevermind, I love the challenge so was fun to workaround! )

I came up with the following which I'm still expanding on to add error handling for missing schedules etc. but works 99% of the time, so included the PowerShell script and the two csv files it uses to allow us to dictate which is a primary and secondary server, and what job categories I want to include.

 Have attached my WIP code here for anyone in the future might hit a similar issue or just to see how bad my code is!
[AgentJob_Login_Backup_v1.0.zip](https://github.com/dataplat/dbatools/files/9196597/AgentJob_Login_Backup_v1.0.zip)

Guessing that no-one else really has an issue with copy-DbaAgentJob and the disable after copy, so don't really expect it to be something that gets changed, especially in a product that is provided for free hoping people provide sponsorship
